apps/dashboard/src/components/workflow-editor/steps/conditions/edit-step-conditions-form.tsx

```diff
@@ -0,0 +1,155 @@
+]);
+
+type FormQuery = {
+  query: RuleGroupType;
+};
+
+const conditionsSchema: z.ZodType<FormQuery> = z.object({
+  query: z
+    .object({
+      combinator: z.string(),
+      rules: z.array(rulesSchema),
+    })
+    .passthrough(),
+});
+
+export const EditStepConditionsForm = () => {
+  const { workflow, step, update } = useWorkflow();
+  const query = useMemo(
+    () =>
+      // prepareRuleGroup and parseJsonLogic calls are needed to generate the unique ids on the query and rules,
+      // otherwise the lib will do it and it will result in the form being dirty
+      step?.controls.values.skip
+        ? prepareRuleGroup(parseJsonLogic(step.controls.values.skip as RQBJsonLogic))
+        : prepareRuleGroup({ combinator: 'and', rules: [] }),
+    [step]
+  );
+  const form = useForm<FormQuery>({
+    mode: 'onSubmit',
+    resolver: zodResolver(conditionsSchema),
+    defaultValues: {
+      query,
+    },
+  });
+  const { formState } = form;
+  const blocker = useBlocker(formState.isDirty);
+  useBeforeUnload(formState.isDirty);
+
+  const { fields, variables } = useMemo(() => {
+    if (!step) return { fields: [], variables: [] };
+
+    const parsedVariables = parseStepVariables(step.variables);
+    return {
+      fields: parsedVariables.primitives.map((primitive) => ({
+        name: primitive.label,
+        label: primitive.label,
+        value: primitive.label,
+      })),
+      variables: [...parsedVariables.primitives, ...parsedVariables.namespaces],
+    };
+  }, [step]);
+
+  const onSubmit = (values: z.infer<typeof conditionsSchema>) => {
+    if (!step || !workflow) return;
+
+    const skip = formatQuery(values.query, 'jsonlogic');
+    const updateStepData: Partial<StepUpdateDto> = {
+      controlValues: { ...step.controls.values, skip },
+    };
+
+    update(updateStepInWorkflow(workflow, step.stepId, updateStepData));
+    form.reset(values);
+  };
+
+  return (
+    <>
+      <Form {...form}>
+        <form className="flex h-full flex-col" onSubmit={form.handleSubmit(onSubmit)}>
+          <div className="flex flex-col gap-3 overflow-y-auto overflow-x-hidden px-3 py-5">
+            <Panel className="overflow-initial">
+              <PanelHeader>
+                <RiInputField className="text-feature size-4" />
+                <span className="text-neutral-950">Step conditions for — {step?.name}</span>
+              </PanelHeader>
+              <PanelContent className="flex flex-col gap-2 border-solid">
+                <FormField
+                  control={form.control}
+                  name="query"
+                  render={({ field }) => (
+                    <ConditionsEditor
+                      query={field.value}
+                      onQueryChange={field.onChange}
+                      fields={fields}
+                      variables={variables}
+                    />
+                  )}
+                />
+              </PanelContent>
+            </Panel>
+            <Link
```

Comment:
I suggest we reuse the layout and the common parts between the step conditions form and the skeleton.